### DISCUSSION TITLE
D1 mini restart loop when connected to an MQTT broker (mosquito) [solved]

### DISCUSSION THREAD

**Discussion by metko1** [arendst/Tasmota#20185] [Q&A]
I've seen several posts across the internet with the problem of restart cycles in the context of MQTT. And I did read the section on MQTT and retained messages in the Tasmota documentation. 
I may be wrong, but for two reasons I believe that this is not the cause of  my problem:
1) Using MQTT Explorer, I searched for (and cleared) all messages from/to my device(s).
2) No matter which prefix I choose in the Tasmota MQTT configuration, the problem persists. My idea is that with a new prefix, there can't be a retained message on the broker.
3) My understanding of the POWER command (retained or entered at the console) is to get or set the status of the relay outputs. How could this lead to a reset?

Now, the log shows cycles of the following messages:

00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: aus Flash geladen von F7, Anzahl 13
00:00:00.055 QPC: Count 1
00:00:00.076 Projekt tasmota - Tasmota Version 13.2.0(tasmota)-2_7_4_9(2023.10.19 09:07:29)
00:00:00.566 WIF: verbinden mit AP1 <XXX>  Channel 11 BSSId 3C:A6:2F:47:A0:38 in Modus 11n als tasmota-7A95B4-5556...
00:00:01.754 WIF: verbunden
00:00:02.006 HTP: Web-Server aktiv bei tasmota-7A95B4-5556 mit IP-Adresse <XXX>
18:04:34.045 MQT: Verbindungsversuch...
18:04:34.268 MQT: verbunden
18:04:34.272 MQT: tele/tasmota_7A95B4/LWT = Online (beibehalten)
18:04:34.274 MQT: cmnd/tasmota_7A95B4/POWER = 
18:04:34.279 MQT: tele/tasmota_7A95B4/INFO1 = {"Info1":{"Module":"Sonoff Basic","Version":"13.2.0(tasmota)","FallbackTopic":"cmnd/DVES_7A95B4_fb/","GroupTopic":"cmnd/tasmotas/"}}
18:04:34.289 MQT: tele/tasmota_7A95B4/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-7A95B4-5556","IPAddress":"192.168.178.55"}}
18:04:34.301 MQT: tele/tasmota_7A95B4/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":6}}
18:04:34.312 MQT: stat/tasmota_7A95B4/RESULT = {"POWER":"OFF"}
18:04:34.316 MQT: stat/tasmota_7A95B4/POWER = OFF
18:04:34.528 MQT: stat/tasmota_7A95B4/RESULT = {"Restart":"Restarting"}
18:04:35.526 APP: starte neu

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 3584, room 16 
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

... and then the whole procedure is repeating. The part I don't understand is how the system comes from a status message  (stat/tasmota_7A95B4/POWER = OFF) to the conclusion it should restart (stat/tasmota_7A95B4/RESULT = {"Restart":"Restarting"}).

Background: The MQTT broker is the public test.mosquitto.org, and the problem began after an outage (and probably a reconfiguration) of the server. I have 6 devices all showing the same behaviour since then. So chances are that the problem is on the server, but maybe somebody has an idea for how the figure out what exactly is going wrong here?

**Comment by sfromis** [comment]
Using that public broker means that you are **volunteering** to be messed with. Cannot be recommended if you favor having control yourself.

If you set `Weblog 3` for details, commands executed should get an indication of the source, like `SRC: MQTT` if arriving via the subscription to the commands topics.

**Reply by metko1** [reply]
Now I configured Tasmota (TLS) to connect to my MQTT account on HiveMQ, and I see the following traffic in the console:
00:00:00.001 HDW: ESP8266EX
00:00:00.054 CFG: Loaded from flash at F7, Count 36
00:00:00.060 QPC: Count 1
00:00:00.081 Project tasmota - Tasmota Version 13.3.0.1(tasmota)-2_7_4_9(2023-12-19T15:02:10)
00:00:00.570 WIF: Connecting to AP1 XXX Channel 1 BSSId 3C:A6:2F:47:A0:38 in mode 11n as tasmota-7A95B4-5556...
00:00:01.754 WIF: Connected
00:00:02.005 HTP: Web server active on tasmota-7A95B4-5556 with IP address XXX
20:06:04.009 MQT: Attempting connection...
20:06:05.025 MQT: TLS connected in 1010 ms, max ThunkStack used 4596
20:06:05.026 MQT: MFLN not supported by TLS server
20:06:05.027 MQT: Connected
20:06:05.103 MQT: tele/tasmota_7A95B4/LWT = Online (retained)
20:06:05.117 MQT: cmnd/tasmota_7A95B4/POWER = 
20:06:05.236 MQT: tele/tasmota_7A95B4/INFO1 = {"Info1":{"Module":"Sonoff Basic","Version":"13.3.0.1(tasmota)","FallbackTopic":"cmnd/DVES_7A95B4_fb/","GroupTopic":"cmnd/tasmotas/"}}
20:06:05.268 MQT: tele/tasmota_7A95B4/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-7A95B4-5556","IPAddress":"1XXX"}}
20:06:05.297 MQT: tele/tasmota_7A95B4/INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":17}}
20:06:05.323 MQT: stat/tasmota_7A95B4/RESULT = {"POWER":"OFF"}
20:06:05.351 MQT: stat/tasmota_7A95B4/POWER = OFF
20:06:07.392 QPC: Reset
20:06:08.396 MQT: tele/tasmota_7A95B4/STATE = {"Time":"2023-12-19T20:06:08","Uptime":"0T00:00:09","UptimeSec":9,"Heap":21,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER":"OFF","Wifi":{"AP":1,"SSId":"XXX","BSSId":"XXX","Channel":1,"Mode":"11n","RSSI":92,"Signal":-54,"LinkCount":1,"Downtime":"0T00:00:03"}}

So, the first part looks the same like for the connection to mosquitto (see first post) up to the point 

20:06:05.351 MQT: stat/tasmota_7A95B4/POWER = OFF

which is followed by a 

20:06:07.392 QPC: Reset

on HiveMQ and by a

18:04:34.528 MQT: stat/tasmota_7A95B4/RESULT = {"Restart":"Restarting"}

on mosquitto. Well, the main difference between the two brokers is that mosquitto permits retained messages wheres HiveMQ does not.  But it's beyond my imagination why the same 

20:06:05.117 MQT: cmnd/tasmota_7A95B4/POWER = 

command leads to a simple QPC:Reset if it is not retained and a full reboot if it is.
Any ideas?

**Reply by barbudor** [reply]
There is no link between Power and QPC Reset

As far as I read that thread you haven't provided a log with level 3 in the condition where the restart occurs

**Reply by sfromis** [reply]
`QPC: Reset` is not interesting to see, other than it means that the boot was successful enough for Tasmota to break the Quick Power Cycle counting for this recovery feature. Ignore this message.

 And if you want more details about what happens do follow the suggestion of a higher log level.

And that initial `MQT: cmnd/tasmota_7A95B4/POWER =` is **not** a command but an artifact produced after boot for backwards compatibility. It does not indicate that "anything happened". Ignore this message.

**Reply by metko1** [reply]
Thanks for these interesting insights. The statement about the POWER command makes sense, likewise does the request for a higher loglevel. Alas, the logs I showed above were obtained with WebLog and SysLog at full debugging output. Strangely my Tasmota does not know MQTTLog - 'unknown command'. Can you confirm? Or were you thinking about other logs?

**Reply by sfromis** [reply]
No, the logs you provided does not include "full debugging output". In the 2nd go, you mentioned getting this line:
`18:04:34.528 MQT: stat/tasmota_7A95B4/RESULT = {"Restart":"Restarting"}`
but you quoted this in isolation, without any context from the log before that. And the very dissimilar timestamps indicate that is is not even from the same boot.

The command `MqttLog` should be available if you have MQTT enabled, but again you posted no log details to verify such. And I was not thinking about using MQTT logging, usually the console should be ok, but the latency for updating the web UI console does mean that the MQTT logging might be useful, but just **one** of WebLog/Syslog/MqttLog/SerialLog should be fine, as long as the output gets used to debug what happens.

**Comment by metko1** [comment]
The problem originated indeed from a retained Reset 2 MQTT message on the mosquitto server as pointed out in the MQTT troubleshooting guide. However, in addition to a Reset command in the cmnd/tasmota_XXXXXX topic (which I had removed), there was another one in the *group topic* cmnd/tasmotas. After deleting this too, everyhing worked again perfectly. Maybe it's worth adding a hint to the group topic feature in the MQTT troubleshooting guide.
